### ISSUE TITLE
Automate the discovery and recovery of Mongo connections

### ISSUE THREAD

**Issue by brianghig** [bluehalo/node-rest-starter#165]
### Background

When a Mongo connection is unavailable on startup, the node app will exit because it rejects the promise waiting for a database connection and throws that error up to the server. This prevents the app from starting and forces an external service to handle restart attempts until the app is able to connect to its database(s).

On the other hand, when a Mongo connection is available on startup, but that connection is then lost, the app is unable to recover the connection, requiring an automated or manual restart of the service.

These connections are all managed in https://github.com/Asymmetrik/node-rest-starter/blob/develop/src/lib/mongoose.js

### Recommended Changes

1. Allow the node app to wait for database connections on initial startup, across all database connections.
2. When a database connection is lost, attempt to recover it by periodically trying to re-establish the connection until it is restored. This interval should be configurable with a reasonable default.
3. Each database connection should be managed in isolation, so outages for one do not impact others. For example, if a secondary "data" connection is unavailable, users should still be able to authenticate through the primary "admin" connection.

**Comment by jrassa** [comment]
> 1. Allow the node app to wait for database connections on initial startup, across all database connections.
I found an example [here](https://stackoverflow.com/a/51998540) of an approach to retrying the initial connection.  It could likely be adapted to retry multiple connections.

> 2\. When a database connection is lost, attempt to recover it by periodically trying to re-establish the connection until it is restored. This interval should be configurable with a reasonable default.

By default, the mongo driver is configured to auto reconnect.  There are 3 options available to configure this behavior: `autoReconnect`, `reconnectInterval`, and `reconnectTries`.

Details on these can be found [here](http://mongodb.github.io/node-mongodb-native/3.0/reference/connecting/connection-settings/)

I just did a quick test and confirmed that it automatically reconnected after stoping and restarting my mongo server.
